packages/legend-extension-external-language-morphir/src/ELMorphir_GenerationPlugin.tsx

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import packageJson from '../package.json';
+
+import type {
+  DSL_StudioPlugin_Extension,
+  GenerationFile,
+  FileGenerationState,
+  StudioPluginManager,
+  FileGenerationResultViewerAction,
+  FileGenerationScopeFilter,
+} from '@finos/legend-studio';
+
+import { StudioPlugin } from '@finos/legend-studio';
+import { NetworkClient, assertNonEmptyString } from '@finos/legend-shared';
+import type { PackageableElement } from '@finos/legend-graph';
+import { ConcreteFunctionDefinition } from '@finos/legend-graph';
+
+const MORPHIR_TYPE_NAME = `morphir`;
+
+interface ELMorphir_GenerationPluginConfigData {
+  morphirVisualizerUrl: string;
+  linterServerUrl: string;
+  linterAppUrl: string;
+}
+
+export class ELMorphir_GenerationPlugin
+  extends StudioPlugin
+  implements DSL_StudioPlugin_Extension
+{
+  networkClient: NetworkClient;
+  morphirVisualizerUrl = ``;
+  linterServerUrl = ``;
+  linterAppUrl = ``;
+
+  constructor() {
+    super(packageJson.extensions.studioPlugin, packageJson.version);
+    this.networkClient = new NetworkClient();
+  }
+
+  install(pluginManager: StudioPluginManager): void {
+    pluginManager.registerStudioPlugin(this);
+  }
+
+  override configure(_configData: object): ELMorphir_GenerationPlugin {
+    const configData = _configData as ELMorphir_GenerationPluginConfigData;
+    assertNonEmptyString(
+      configData.morphirVisualizerUrl,
+      `Can't configure morphir visualizer url for generation plugin: 'url' field is missing or empty`,
+    );
+    assertNonEmptyString(
+      configData.linterServerUrl,
+      `Can't configure linter server url for generation plugin: 'url' field is missing or empty`,
+    );
+    assertNonEmptyString(
+      configData.linterAppUrl,
+      `Can't configure linter app url for generation plugin: 'url' field is missing or empty`,
+    );
+    this.morphirVisualizerUrl = configData.morphirVisualizerUrl;
+    this.linterServerUrl = configData.linterServerUrl;
+    this.linterAppUrl = configData.linterAppUrl;
+
+    return this;
+  }
+
+  visualizeMorphir =
```

Comment:
move these `visualize...` functions to inside the method that renders that buttons

Reply:
Fixed.